crates/rooch-rpc-client/src/lib.rs

```diff
@@ -50,9 +58,24 @@ impl ClientBuilder {
                 .build(http)?,
         );
 
+        if let Some(proxy_url) = self.proxy_url {
+            let reqwest_client = Some(Arc::new(
+                reqwest::Client::builder()
+                    .proxy(Proxy::https(proxy_url.clone())?)
+                    .proxy(Proxy::all(proxy_url)?)
+                    .build()?,
+            ));
+            return Ok(Client {
+                http: http_client.clone(),
+                rooch: RoochRpcClient::new(http_client.clone()),
+                reqwest: reqwest_client,
```

Comment:
We should find method to add proxy to HTTP client and replace the HTTP client in RoochRpcClient, not add a new client.

Reply:
> We should find method to add proxy to HTTP client and replace the HTTP client in RoochRpcClient, not add a new client.

From my research, the HTTP client doesn't support proxied request. However, I'll find a way to add a proxy to the http client.

Reply:
From what I could find so far, the `JsonRpsee` lib only supports `socks5` proxy, meaning that there's no way to connect with `https_proxy` or `all_proxy` through using only the `JsonRpsee` lib. I'm sorry for that.

1. https://grants.web3.foundation/docs/RFPs/jsonrpsee-proxy-support
2. https://github.com/paritytech/jsonrpsee/issues/870
3. https://github.com/paritytech/jsonrpsee/pull/1168